dubbo-configcenter/dubbo-configcenter-redis/src/main/java/org/apache/dubbo/configcenter/support/redis/RedisDynamicConfiguration.java

```diff
@@ -0,0 +1,177 @@
+            redisClient = new SentinelRedisClient(url);
+        } else if (CLUSTER_REDIS.equals(type)) {
+            redisClient = new ClusterRedisClient(url);
+        } else {
+            redisClient = new MonoRedisClient(url);
+        }
+        long checkPassInterval = url.getParameter(CHECK_CONFIG_INTERVAL, DEFAULT_CHECK_CONFIG_INTERVAL);
+        configCheckExecutor = new ScheduledThreadPoolExecutor(1, new NamedThreadFactory("Ttl-Redis-Check-Executor", true));
+        configCheckExecutor.scheduleAtFixedRate(this::checkConfig, checkPassInterval / 8,
+                checkPassInterval / 8, TimeUnit.MILLISECONDS);
+    }
+
+    @Override
+    protected boolean doPublishConfig(String pathKey, String content) throws Exception {
+        return OK_RESPONSE.equals(redisClient.set(pathKey, content));
+    }
+
+    @Override
+    protected String doGetConfig(String pathKey) throws Exception {
+        return redisClient.get(pathKey);
+    }
+
+    @Override
+    protected boolean doRemoveConfig(String pathKey) throws Exception {
+        redisClient.del(pathKey);
+        return true;
+    }
+
+    @Override
+    protected Collection<String> doGetConfigKeys(String groupPath) {
+        if (!groupPath.endsWith(ANY_VALUE)) {
+            if (!groupPath.endsWith(SLASH)) {
+                groupPath = groupPath + SLASH;
+            }
+            groupPath = groupPath + ANY_VALUE;
+        }
+        return redisClient.scan(groupPath);
+    }
+
+    @Override
+    protected void doAddListener(String pathKey, ConfigurationListener listener) {
+        logger.info("register listener " + listener.getClass() + " for config with key: " + pathKey);
+        String value = redisClient.get(pathKey);
+        if (value == null) {
+            configHashes.put(pathKey, HashHolder.getNullHash());
+        } else {
+            configHashes.put(pathKey, new HashHolder(value.hashCode()));
+        }
+        configListeners.put(pathKey, listener);
+    }
+
+    @Override
+    protected void doRemoveListener(String pathKey, ConfigurationListener listener) {
+        logger.info("unregister listener " + listener.getClass() + " for config with key: " + pathKey);
+        configHashes.remove(pathKey);
+        configListeners.remove(pathKey);
+    }
+
+    @Override
+    protected void doClose() throws Exception {
+        configHashes.clear();
+        configListeners.clear();
+        configCheckExecutor.shutdownNow();
+        redisClient.close();
+    }
+
+    private void checkConfig() {
+        for (Map.Entry<String, HashHolder> entry : configHashes.entrySet()) {
+            String value = redisClient.get(entry.getKey());
+            if (value != null) {
+                Integer hash = Integer.valueOf(value.hashCode());
+                Integer originHash = entry.getValue().getHash();
+                if (!hash.equals(originHash)) {
+                    System.out.println("-------------key: " + entry.getKey() + ", -----------value: " + entry.getValue());
```

Comment:
I think using `System.io` to output is need to banned. pls use `logger` instead.

Reply:
Thanks! It is my test code, and I forgot to delete.